### PR TITLE
GCI: Shift From Sugar ORM to Room

### PR DESCRIPTION
**Pull request by HITGIF**
For the [task](https://codein.withgoogle.com/dashboard/task-instances/6680678198411264/).
As the feature was implemented already and further changes had been made base on that, this PR is not meant to be merged.

### CODE REVIEW

**Comment by m-murad** [comment]
Please resolve the conflict

**Comment by HITGIF** [comment]
Conflicts resolved.

File: app/src/main/java/com/community/jboss/leadmanagement/main/contacts/ContactsAdapter.java [hunk 1]
```diff
@@ -1,239 +1,110 @@
...
+            }
+        });
     }
-}
 
+    @Override
+    public int getItemCount() {
+        return contacts.size();
+    }
+}
```
Comment (aashutoshrathi) [inline app/src/main/java/com/community/jboss/leadmanagement/main/contacts/ContactsAdapter.java]: Always add newlines at the end

File: app/src/main/java/com/community/jboss/leadmanagement/main/contacts/ContactsFragment.java [hunk 2]
```diff
@@ -1,123 +1,78 @@
...
-        }
-        else {
-            textView.setVisibility(View.GONE);
-        }
-        return true;
+    public int getTitle() {
+        return R.string.title_contacts;
     }
-}
+}
```
Comment (aashutoshrathi) [inline app/src/main/java/com/community/jboss/leadmanagement/main/contacts/ContactsFragment.java]: Add newline

File: app/src/main/java/com/community/jboss/leadmanagement/AddContactActivity.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            if(currentContact.getName().equals(_name) && currentContactNumber.getNumber().equals(_number)){
+                this.finish();
+            }else if (currentContactNumber.getNumber().length()>0) {
+                currentContact.setName(_name);
+                leadDB.getContactDao().update(currentContact);
+                currentContactNumber.setNumber(_number);
+                leadDB.getContactNumberDao().update(currentContactNumber);
+                this.finish();
+            }
+        }else {
```
Comment (aashutoshrathi) [inline app/src/main/java/com/community/jboss/leadmanagement/AddContactActivity.java]: space before else

File: app/src/main/java/com/community/jboss/leadmanagement/AddContactActivity.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        final String number = intent.getStringExtra(INTENT_EXTRA_NUMBER);
+
+        if(leadDB.getContactNumberDao().getContactNumber(number)!=null){
+            this.currentContactNumber = leadDB.getContactNumberDao().getContactNumber(number);
+            setTitle(R.string.contact_edit);
+            contactNameField.setText(currentContactNumber != null ?
+                    leadDB.getContactDao().getContact(currentContactNumber.getContactId()).getName() : "");
+            contactNumberField.setText(number);
+        }else{
```
Comment (aashutoshrathi) [inline app/src/main/java/com/community/jboss/leadmanagement/AddContactActivity.java]: space before else

**Comment by HITGIF** [comment]
Formatted & squashed :)

**Comment by HITGIF** [comment]
PLEASE don't merge this one...
This PR is implementing an old functionality and is thus based on the branch from an old commit (the commit for merging #48 specifically).
This PR is not meant to be merged but only for the purpose of learning and GCI, merging it will result in the override of all the changes made after merging #48, which is pretty bad.
Revert this one please :)

**Pull request merged**